components/raftstore/src/store/async_io/write_router.rs

```diff
@@ -238,39 +243,90 @@ where
                 let now = Instant::now();
                 if sender.send(msg, self.last_msg_priority).is_err() {
                     // Write threads are destroyed after store threads during shutdown.
-                    panic!("{} failed to send write msg, err: disconnected", self.tag);
+                    safe_panic!("{} failed to send write msg, err: disconnected", self.tag);
                 }
                 ctx.raft_metrics()
                     .write_block_wait
                     .observe(now.saturating_elapsed_secs());
             }
             Err(TrySendError::Disconnected(_)) => {
                 // Write threads are destroyed after store threads during shutdown.
-                panic!("{} failed to send write msg, err: disconnected", self.tag);
+                safe_panic!("{} failed to send write msg, err: disconnected", self.tag);
             }
         }
     }
 }
 
+#[derive(Clone)]
+/// Safefly shared senders among the controller and raftstore threads.
+/// Senders in it can only be accessed by cloning method `senders()`.
+
```

Comment:
Remove the newline, and put comment above `derive`, and comment that `Clone` is safe to race with concurrent `send()` because `RefCell` field `last_msg_group` is skipped, and leave a TODO to avoid this hack using Mutex.